### PR TITLE
feature/pratyush1718/companyRoutes

### LINKED ISSUE #22: Backend - Create companies routes
- Create the following routes to interact with the "users" collection
  - "/addCompany"
    - Post
    - Require a company name, email, and password 
  - "/editCompany/:id"
    - Put
    - Can change company name, email, and/or password
  - "/deleteCompany/:id"
    - Delete
    - Delete company based on id
  - "/getCompany/:id"
    - Get
    - Gets company on id
  - "/getAllCompanies"
    - Get
    - Gets all companies

### PR DESCRIPTION
**Pull request by pratyush1718**
### Administrative Info
[GitHub Issue #]
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Added company routes. Also fixed arrow function bug in Users schema

### Testing
How did you confirm your changes worked? 
- Ran all routes through postman

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.
DELETE_Companies
GETall_Companies
GET_Companies
POST_Companies
PUT_Companies

### CODE REVIEW

**Review by navidboloorian — APPROVED** [review]
LGTM!

File: backend/models/Users.js [hunk 1]
```diff
@@ -55,11 +55,14 @@ UserSchema.pre("findOneAndUpdate", async function (next) {
   }
 });
 
-UserSchema.methods.verifyPassword = (enteredPassword) =>
+UserSchema.methods.verifyPassword = function (enteredPassword) {
+  console.log(enteredPassword);
+  console.log(this.password);
   bcrypt.compare(enteredPassword, this.password, (err, matchBool) => {
```
Comment (wllmwu) [inline backend/models/Users.js:61]: Using `function () {}` syntax followed by arrow function `() => {}` syntax. Not really an issue but try to be consistent in the future

File: backend/routes/companiesRoutes.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
+/**
+ * API Routes for Companies Schema
+ * POST (/addCompany)
+ * DELETE (/deleteCompany/:id)
+ * PUT (/editCompany/:id)
+ * GET (/getCompany/:id)
+ * GET (/getAllCompanies)
```
Comment (wllmwu) [inline backend/routes/companiesRoutes.js:7]: These routes could be made more RESTful. For example,
```
POST /companies
DELETE /companies/:id
PUT /companies/:id
GET /companies/:id
GET /companies
```
The same goes for other routes, like the ones for users. In a RESTful API, each route's "action" is determined by the HTTP verb (GET = get, POST = create, etc.), so you don't need to repeat that in the route's path. And the routes are kind of named in terms of the resource, e.g. these routes deal with companies so they all have `/companies`.

This is just a housekeeping issue, doesn't affect the functionality, but I'm sure Justin would call it out during code quality checks.

**Review by wllmwu — CHANGES_REQUESTED** [review]
I noticed our routes don't seem to be RESTful. @navidboloorian your call on whether we should refactor them or not, since they've been this way for a while. If not then I'll approve 👍

File: backend/models/Users.js (hunk 1 above)
Comment (navidboloorian) [reply backend/models/Users.js:61]: There've been some issues with this context with arrow function syntax. Good call to use function syntax as far as these backend issues are concerned.

File: backend/routes/companiesRoutes.js (hunk 2 above)
Comment (navidboloorian) [reply backend/routes/companiesRoutes.js:7]: I actually wasn't aware of this standard. @pratyush1718 please implement these changes in your route files. Going forward I'll be cognizant of this.

**Review by wllmwu — APPROVED** [review]
Great work! Hopefully this will please our overlord Justin.

**Pull request merged**